src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts

```diff
@@ -256,7 +262,12 @@ export class SolutionExplorerList {
 
     const solutionIsNotOpened: boolean = viewModelOfEntry === undefined || viewModelOfEntry === null;
     if (solutionIsNotOpened) {
-      await this.openSolution(uri);
+      if (uri === 'Single Diagrams') {
```

Comment:
Extract Business Rule
```suggestion
      const uriIsSingleDiagrams: boolean = uri === 'Single Diagrams';
      if (uriIsSingleDiagrams) {
```